src/nl/rubensten/texifyidea/settings/labeldefiningcommands/EditLabelDefiningCommand.kt

```diff
@@ -0,0 +1,104 @@
+ */
+class EditLabelDefiningCommand(cmdName: String, position: Int, labelAnyPrevCommandStatus: Boolean) :
+        DialogWrapper(null, false) {
+    private val commandName: JTextField = JTextField(cmdName, 25)
+    private val commandPosition: JTextField = JTextField("$position", 5)
+    private val labelAnyPrevCommand: JCheckBox = JCheckBox("Labels a previous command like \\section",
+            labelAnyPrevCommandStatus)
+
+    init {
+        title = "Define command with label"
+        (commandName.document as PlainDocument).documentFilter = InputCommandFilter()
+        (commandPosition.document as PlainDocument).documentFilter = InputPossiblePositionFilter()
+        super.init()
+    }
+
+    override fun createCenterPanel(): JComponent? = null
+
+    /**
+     * create the layout of the window
+     */
+    override fun createNorthPanel(): JComponent? {
+        val remoteComponent = JPanel(GridBagLayout())
+        val gridBag = GridBagConstraints()
+        gridBag.fill = GridBagConstraints.HORIZONTAL
+        gridBag.gridx = 0
+        gridBag.gridy = 0
+        remoteComponent.add(JBLabel("Name: ", SwingConstants.RIGHT), gridBag)
+        gridBag.gridx = 1
+        remoteComponent.add(commandName, gridBag)
+        gridBag.gridx = 0
+        gridBag.gridy = 1
+        remoteComponent.add(JBLabel("Position: ", SwingConstants.RIGHT), gridBag)
+        gridBag.gridx = 1
+        remoteComponent.add(commandPosition, gridBag)
+        gridBag.gridx = 1
+        gridBag.gridy = 2
+        val label = JLabel("Only required parameters count for the position", SwingConstants.LEFT)
+        label.foreground = Color.GRAY
+        remoteComponent.add(label, gridBag)
+
+        gridBag.gridx = 1
+        gridBag.gridy = 3
+        gridBag.gridwidth = 2
+        remoteComponent.add(labelAnyPrevCommand, gridBag)
+
+        peer.window.addWindowListener(object : WindowAdapter() {
+            override fun windowOpened(e: WindowEvent?) {
+                commandName.requestFocus()
+            }
+        })
+
+        return remoteComponent
+    }
+
+    /**
+     * validate the input so it is not empty and the Int value is in possible range
+     */
+    override fun doValidateAll(): MutableList<ValidationInfo> {
+        val list = mutableListOf<ValidationInfo>()
+        if (commandName.text.trim().isEmpty()) {
+            list.add(ValidationInfo("Name can not be empty"))
+        }
+        if (!commandPosition.text.trim().possiblePosition()) {
+            list.add(ValidationInfo("Position must be convertible to Int"))
+        }
+        return list
+    }
+
+    /**
+     * return the name of the command
+     */
+    fun getCommandName(): String {
+        return commandName.text
+    }
+
+    /**
+     * return the position of the label
+     */
+    fun getCommandPosition(): Int {
+        return commandPosition.text.toInt()
+    }
+
+    /**
+     * return whether the command labels any previous command or not
+     */
```

Comment:
either unnecessary, or wrongly documented

Reply:
fixed in commit ed34f48cd6dd8da4a8ed3d61d6cf44599f4bc7dd